enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -244,89 +258,37 @@ complete OpenShift releases: blobs take 16 GiB and when they are written to disk
 they take 32 GiB.
 
 If the calculate disk space exceeds the available disk space then the
-`PinnedImageSet` will be marked as failed and the process will not move forward.
-For example:
-
-```yaml
-status:
-  conditions:
-  - type: Ready
-    status: "False"
-  - type: Failed
-    status: "True"
-    message: |
-      Pulling the images in `node12` requires at least 16 GiB of disk
-      space, but there are only 10 GiB available
-```
+`prefetch_image_manger` will report the error as a `Degraded` = `true` status
+for the MachineConfigPool with the error in the `message`.
 
 Note that even with this check it will still be possible (but less likely) to
 have failures to pull images due to disk space: the heuristic could be wrong,
 and there may be other components pulling images or consuming disk space in
 some other way. Those failures will be detected and reported in the status of
-the `PinnedImageSet` when CRI-O fails to pull the image.
-
-The steps above will happen in all the nodes of the cluster. The daemon set
-will be provided with enough information to ensure that each pod applies only
-the changes required for the node where it runs, according to the node
-selectors in the `PinnedImageSet` custom resources.
+the `MachineConfigPool` when CRI-O fails to pull the image.
 
 When all the images have been successfully pinned and pulled in all the matching
-nodes the `PinnedImageSetController` will set the `Ready` condition to `True`:
+nodes the `MachineConfigPool` will use `nodeWriter.SetDone()` to communicate to
+the pool that work is done.
 
-```yaml
-status:
-  conditions:
-  - type: Ready
-    status: "True"
-  - type: Failed
-    status: "False"
-```
+Today the Updated status reflects only the rendered worker config. The message
+will need to be slightly updated in the case where prefetch is a success.
 
-If something fails the `Failed` condition will be set to `True`, and the
-details of the error will be in the message. For example if `node12` fails to
-pull an image:
-
-```yaml
-status:
-  pinnedImages:
-  - quay.io/openshift-release-dev/ocp-release@sha256:...
-  - quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:...
-  conditions:
-  - type: Ready
-    status: "False"
-  - type: Failed
-    status: "True"
-    message: Node 'node12' failed to pull image `quay.io/...` because ...
 ```
-
-We will consider using the new _machine-config-operator_ state reporting
-mechanism introduced [https://github.com/openshift/api/pull/1596](here) to
-report additional details about the progress or issues of each node.
-
-Additional information may be needed inside the `status` field of the
-`PinnedImageSet` custom resources in order to implement the mechanisms described
-above. For example, it may be necessary to have conditions per node, something
-like this:
-
-```yaml
-status:
-  node0:
-  - type: Ready
-    status: "False"
-  - type: Failed
+  - lastTransitionTime: "2024-03-14T20:31:32Z"
```

Comment:
I think reason is mandatory, not a big deal but we should aim to have that as well as a message.

Reply:
I took this from a 4.16 cluster :)